process/smartContract/process.go

```diff
@@ -1454,6 +1523,34 @@ func (sc *scProcessor) createSmartContractResults(
 	return createdAsyncCallBack, scResults
 }
 
+func (sc *scProcessor) useLastTransferAsAsyncCallBackWhenNeeded(
+	callType vmcommon.CallType,
+	outAcc *vmcommon.OutputAccount,
+	outputTransfer *vmcommon.OutputTransfer,
+	vmOutput *vmcommon.VMOutput,
+	tx data.TransactionHandler,
+	result *smartContractResult.SmartContractResult,
+) bool {
+	if len(vmOutput.ReturnData) > 0 {
+		return false
+	}
+
+	isAsyncTransferBackToSender := callType == vmcommon.AsynchronousCall &&
+		bytes.Equal(outAcc.Address, tx.GetSndAddr())
+	if !isAsyncTransferBackToSender {
+		return false
+	}
+
+	if !sc.isTransferWithNoDataOrBuiltInCall(outputTransfer.Data) {
+		return false
+	}
+
+	result.CallType = vmcommon.AsynchronousCallBack
+	result.GasLimit += vmOutput.GasRemaining
```

Comment:
safeAdd?

Reply:
error was treated before on consumedGasChecks which is called before this function.